docs/plugins/haveibeenpwned/plugin.js

```diff
@@ -0,0 +1,250 @@
+DetailsView.prototype.blockPwnedPwd = false;
+DetailsView.prototype.blockPwnedName = false;
+DetailsView.prototype.logger = new Logger("HaveIBeenPwned");
+
+DetailsView.prototype._alert = function (msg) {
+    Alerts.info({ body: msg, title: "HaveIBeenPwned" });
+}
+
+DetailsView.prototype.checkPwnedOnSettingsChanged = function (changes) {
+    //if (changes['CheckPwnedPwd'] || changes['CheckPwnedName'] || changes['CheckPwnedName'] || changes['CheckPwnedName']) {
+    // info('Full HaveIBeenPwned check not yet implemented. Checks are done one by one when you change a name or a password.');
+    //}
+};
+
+seen = [];
+class HIBPUtils {
+    constructor() {
+        seen = [];
+    }
+
+    replacer(key, value) {
+        if (value != null && typeof value == "object") {
+            if (seen.indexOf(value) >= 0) {
+                return;
+            }
+            seen.push(value);
+        }
+        return value;
+    }
+    stringify(obj) {
+        var ret = JSON.stringify(obj, this.replacer);
+        seen = [];
+        return ret;
+    }
+    xhrcall (config) {
+        const xhr = new XMLHttpRequest();
+        if (config.responseType) {
+            xhr.responseType = config.responseType;
+        }
+        const statuses = config.statuses || [200];
+        xhr.addEventListener('load', () => {
+            if (statuses.indexOf(xhr.status) >= 0) {
+                return config.success && config.success(xhr.response, xhr);
+            } else {
+                return config.error && config.error('http status ' + xhr.status, xhr);
+            }
+        });
+        xhr.addEventListener('error', () => {
+            return config.error && config.error('network error', xhr);
+        });
+        xhr.addEventListener('timeout', () => {
+            return config.error && config.error('timeout', xhr);
+        });
+        xhr.open(config.method || 'GET', config.url);
+        if (config.headers) config.headers.forEach((value, key) => {
+            xhr.setRequestHeader(key, value);
+        });
+        xhr.send(config.data);
+    };
+
+    hex (buffer) {
+        var hexCodes = [];
+        var view = new DataView(buffer);
+        for (var i = 0; i < view.byteLength; i += 4) {
+            // Using getUint32 reduces the number of iterations needed (we process 4 bytes each time)
+            var value = view.getUint32(i)
+            // toString(16) will give the hex representation of the number without padding
+            var stringValue = value.toString(16)
+            // We use concatenation and slice for padding
+            var padding = '00000000'
+            var paddedValue = (padding + stringValue).slice(-padding.length)
+            hexCodes.push(paddedValue);
+        }
+
+        // Join all the hex strings into one
+        return hexCodes.join("");
+    }
+
+    digest(algo, str) {
+        // We transform the string into an arraybuffer.
+        const buffer = new TextEncoder("utf-8").encode(str);
```

Comment:
I'd suggest using `require('kdbxweb').ByteUtils.stringToBytes`, e.g. TextEncoder is not implemented everywhere.

Reply:
done